### PR TITLE
Fixes for building with `WOLFSSL_USER_IO`

### PR DESCRIPTION
**Pull request by dgarske**
* Fixes for building with `WOLFSSL_USER_IO` (with no built-in socket support). Relates to issue #3998.
* Cleanup of the socket response code handling.

### CODE REVIEW

**Comment by toddouska** [comment]
Looks good, @dgarske please resolve merge conflict.

File: src/wolfio.c [hunk 1]
```diff
@@ -47,7 +47,7 @@ Possible IO enable options:
  * WOLFSSL_USER_IO:     Disables default Embed* callbacks and     default: off
                         allows user to define their own using
                         wolfSSL_CTX_SetIORecv and wolfSSL_CTX_SetIOSend
- * USE_WOLFSSL_IO:      Enables the wolfSSL IO functions          default: off
+ * USE_WOLFSSL_IO:      Enables the wolfSSL IO functions          default: on
```
Comment (SparkiDev) [inline src/wolfio.c:50]: Default is not 'on' from what I can see.

Comment (dgarske) [reply src/wolfio.c:50]: The macro `USE_WOLFSSL_IO` is on by default when `WOLFSSL_USER_IO` is not defined.

wolfio.h:43
```
#if !defined(WOLFSSL_USER_IO)
    /* Micrium uses NetSock I/O callbacks in wolfio.c */
    #if !defined(USE_WOLFSSL_IO) && !defined(MICRIUM) && \
        !defined(WOLFSSL_CONTIKI) && !defined(WOLFSSL_NO_SOCK)
```

Comment (SparkiDev) [reply src/wolfio.c:50]: See it now.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**